create basic javascript project skeleton

benbieler opened the pull request:
- configure grunt runner
- import dependencies
- resolves #2

benbieler committed:
Task #2 create basic javascript project skeleton

- configure grunt runner
- create package.json
- adding basic test suite using facebook-jest
- resolves #2

Ma27 commented:
see #8 .
This pull request contains the same commit on the right branch

Ma27 commented on .gitignore:
```diff
@@ -36,5 +36,18 @@
+/web/build/
+/src/Frontend/styles/custom.css
+/src/Frontend/styles/custom.css.map
+.sass-cache/
```
there's a / before the directory name missing

Ma27 commented on .gitignore:
```diff
@@ -36,5 +36,18 @@
+# PHPStorm
+/.idea/*
+
+# Frontend build data
```
should be renamed to "Frontend files and tools"

Ma27 commented on Gruntfile.js:
```diff
@@ -0,0 +1,77 @@
+                    'web/build/temp/normalize.css',
+                    'src/Frontend/styles/custom.css'
+                ],
+                dest: 'web/build/all.css'
```
should be renamed to _bundle.css_

Ma27 commented on package.json:
```diff
@@ -0,0 +1,41 @@
+  },
+  "keywords": [
+    "Social Network",
+    "Symfony2",
```
this keyword should not be in a frontend configuration file

Ma27 commented on src/Frontend/components/helloWorld.js:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
every source file (the scss files, too) should have the same header as in the old SenNetwork repository: https://github.com/Ma27/SenNetwork/blob/master/src/Sententiaregum/AppBundle/Command/PurgeNotActivatedUsersCommand.php#L3-L10

Ma27 commented:
please fix the comments and implement a simple testsuite using jest (https://facebook.github.io/jest/docs/tutorial-react.html)

Ma27 commented on Gruntfile.js:
```diff
@@ -0,0 +1,90 @@
+    grunt.loadNpmTasks('grunt-contrib-cssmin');
+    grunt.loadNpmTasks('grunt-contrib-watch');
+
+    grunt.registerTask('build', [
```
the build task should contain the tasks _build_css_ and _build_js_

Ma27 commented on Gruntfile.js:
```diff
@@ -0,0 +1,87 @@
+        },
+        browserify: {
+            client: {
+                src: ['src/Frontend/**/*.js'],
```
you have to ignore all files of the test directory in order to prevent the compilation of these: http://stackoverflow.com/questions/29034967/grunt-browserify-ignore-option

Ma27 commented:
:+1: 
@benbieler thank you very much! I'll squash it into one commit and merge it :)

Ma27 committed:
Task #2 install npm dependencies during composer install

- added script handler that processes the npm command using the process
  component
- references #2

Ma27 commented:
thanks a lot for this awesome pull request @benbieler
